src/hooks/useIsEncrypted.ts

```diff
@@ -6,19 +6,28 @@ SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only
 Please see LICENSE files in the repository root for full details.
 */
 
-import { useCallback, useState } from "react";
+import { useCallback, useEffect, useState } from "react";
 import { MatrixClient, MatrixEvent, Room, RoomStateEvent, EventType } from "matrix-js-sdk/src/matrix";
 
 import { useTypedEventEmitter } from "./useEventEmitter";
 
-// Hook to simplify watching whether a Matrix room is encrypted, returns undefined if room is undefined
-export function useIsEncrypted(cli: MatrixClient, room?: Room): boolean | undefined {
-    const [isEncrypted, setIsEncrypted] = useState(room ? cli.isRoomEncrypted(room.roomId) : undefined);
+// Hook to simplify watching whether a Matrix room is encrypted, returns undefined if room is undefined, return null when isRoomEncrypted is computed
```

Comment:
Do we really need both `undefined` and `null`? I'd suggest `null` for `if room is undefined or the state is loading`